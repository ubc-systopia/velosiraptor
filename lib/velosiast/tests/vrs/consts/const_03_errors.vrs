// that's a conflict
const FOO : int = 1234;
const FOO : int = 1234;

// wrong types
const BAR1: bool = 1234;
const BAR2: int = true;
const BAR3 : MyType = 1234;

// undefined value
const FOOBAR : bool = UNDEFINED;
