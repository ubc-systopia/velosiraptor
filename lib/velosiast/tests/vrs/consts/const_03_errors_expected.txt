error: duplicate definitions with name `FOO`
     --> tests/vrs/consts/const_03_errors.vrs:3:1
      |
    2 |  const FOO : int = 1234;
      |  ^^^^^^^^^^^^^^^^^^^^^^^ previous definition of `FOO` was here
   ...
      |
    3 |  const FOO : int = 1234;
      |  ^^^^^^^^^^^^^^^^^^^^^^^ duplicate definition
error: mismatched types
     --> tests/vrs/consts/const_03_errors.vrs:6:20
      |
    6 |  const BAR1: bool = 1234;
      |                     ^^^^ expected `boolean`, found `numeric`
error: mismatched types
     --> tests/vrs/consts/const_03_errors.vrs:7:19
      |
    7 |  const BAR2: int = true;
      |                    ^^^^ expected `numeric`, found `boolean`
error: no symbol named `MyType` found in the current scope
     --> tests/vrs/consts/const_03_errors.vrs:8:14
      |
    8 |  const BAR3 : MyType = 1234;
      |               ^^^^^^ not found in this scope
error: Unsupported type. Constant definitions only support of the built-in types.
     --> tests/vrs/consts/const_03_errors.vrs:8:14
      |
    8 |  const BAR3 : MyType = 1234;
      |               ^^^^^^ Change this type to one of [`bool`, `int`, `size`, `addr`].
error: no symbol named `UNDEFINED` found in the current scope
     --> tests/vrs/consts/const_03_errors.vrs:11:23
      |
   11 |  const FOOBAR : bool = UNDEFINED;
      |                        ^^^^^^^^^ not found in this scope
error: Expected a constant expression.
     --> tests/vrs/consts/const_03_errors.vrs:11:23
      |
   11 |  const FOOBAR : bool = UNDEFINED;
      |                        ^^^^^^^^^ Convert this expression into a constant expression
error: mismatched types
     --> tests/vrs/consts/const_03_errors.vrs:11:23
      |
   11 |  const FOOBAR : bool = UNDEFINED;
      |                        ^^^^^^^^^ expected `boolean`, found `numeric`
error: could not compile due to 8 previous errors; 0 warnings emitted
